tests/e2e/specs/restore-trash-post.test.js

```diff
@@ -0,0 +1,50 @@
+import {
+    visitAdminPage,
+    createNewPost,
+    trashAllPosts,
+    publishPost
+} from '@wordpress/e2e-test-utils';
+
+describe( 'Restore trash post', () => {
+	beforeEach( async () => {
+		await trashAllPosts();
+	} );
+	
+    it( 'displays a message in the posts table when no posts are present', async () => {
+		await visitAdminPage( '/edit.php' );
+		const noPostsMessage = await page.$x(
+			'//td[text()="No posts found."]'
+		);
+		expect( noPostsMessage.length ).toBe( 1 );
+	} );
+
+    it( 'Restore trash post', async () => {
+    
+        //create a Post
+        const title = 'Test Title';
+        await createNewPost( { title } );
+        await publishPost();
+
+        await visitAdminPage( '/edit.php' );  
+
+        // Move one post to trash
+        await page.waitForSelector( '#the-list .type-post' );
+        await page.hover('.row-title');
+        await page.click("a[aria-label='Move “Test Title” to the Trash']");
+
+        // Remove post from trash 
+	await page.click("a[href='edit.php?post_status=trash&post_type=post']");
```

Comment:
Nit: Missing an indention.

```suggestion
		await page.click("a[href='edit.php?post_status=trash&post_type=post']");
```